gluten-substrait/src/test/scala/org/apache/spark/sql/GlutenQueryTest.scala

```diff
@@ -77,31 +91,49 @@ abstract class GlutenQueryTest extends PlanTest {
     shouldRun
   }
 
-  def ignore(
+  /** Ignore the test if the current spark version is between the minVersion and maxVersion */
+  def ignoreWithSpecifiedSparkVersion(
       testName: String,
       minSparkVersion: Option[String] = None,
       maxSparkVersion: Option[String] = None)(testFun: => Any): Unit = {
-    if (shouldRun(minSparkVersion, maxSparkVersion)) {
+    if (matchSparkVersion(minSparkVersion, maxSparkVersion)) {
       ignore(testName) {
         testFun
       }
     }
   }
 
-  def testWithSpecifiedSparkVersion(
-      testName: String,
-      minSparkVersion: Option[String] = None,
-      maxSparkVersion: Option[String] = None)(testFun: => Any): Unit = {
-    if (shouldRun(minSparkVersion, maxSparkVersion)) {
+  /** Run the test if the current spark version is between the minVersion and maxVersion */
+  def testWithRangeSparkVersion(testName: String, minSparkVersion: String, maxSparkVersion: String)(
+      testFun: => Any): Unit = {
+    if (matchSparkVersion(Some(minSparkVersion), Some(maxSparkVersion))) {
       test(testName) {
         testFun
       }
     }
   }
 
-  def testWithSpecifiedSparkVersion(testName: String, versions: Array[String])(
-      testFun: => Any): Unit = {
-    if (versions.exists(v => shouldRun(Some(v), Some(v)))) {
+  /** Run the test if the current spark version less than the maxVersion */
+  def testAtMaxSparkVersion(testName: String, maxVersion: String)(testFun: => Any): Unit = {
+    if (matchSparkVersion(maxSparkVersion = Some(maxVersion))) {
+      test(testName) {
+        testFun
+      }
+    }
+  }
+
+  /** Run the test if the current spark version greater than the minVersion */
+  def testAtLeastSparkVersion(testName: String, minVersion: String)(testFun: => Any): Unit = {
```

Comment:
nit: `testWithMinSparkVersion` and `testWithMaxSparkVersion` sounds better to me.